pkg/controller/certificaterequests/acme/acme.go

```diff
@@ -218,9 +219,15 @@ func buildOrder(cr *v1.CertificateRequest, csr *x509.CertificateRequest) (*cmacm
 		IPAddresses: ipAddresses,
 	}
 
+	if issuer.GetSpec().ACME.EnableNotAfterDate {
+		notAfterTime := metav1.NewTime(time.Now().Add(cr.Spec.Duration.Duration))
```

Comment:
If we need to use the current time, we should be using the `clock.Clock` interface in order to make testing possible.

Also, encoding NotAfter into the Order resource could be problematic in instances where an Order does not get actuated/completed quickly, in which case the certificate being requested by have a NotAfter date that is in the past. How should the ACME certificaterequest controller handle this? Should the order be marked as failed? Some ACME servers may reject the request if it has a NotAfter time in the past, so we should have *some* kind of handling for it :)

Reply:
Not After is called on order create in the ACME spec, weirdly enough... so not entirely sure about how an ACME server will handle that. Thus this being in the spec part of the order, i can deal with order k8s creation and order acme creation drift. Thinking more about it I think we should given orders can be retried thus ending up in an infinite retry

big +1 on the clock!